presentation/common/src/presentation-common/KoqPropertyValueFormatter.ts

```diff
@@ -59,7 +59,7 @@ interface FormattingProps {
   persistenceUnitName: string;
 }
 
-async function getFormattingProps(schemaLocater: ISchemaLocater, options: FormatOptions): Promise<FormattingProps | undefined> {
```

Comment:
I think it was a mistake to use ISchemaLocater here ... in all cases a SchemaContext is used and using a locater here would be bad because it would load the schema every time the method was called instead of using a version that was already loaded

Reply:
I see what you mean, but from API standpoint before your changes, I don't see why these methods couldn't have taken `ISchemaLocater`. All these functions care is to get a schema - they don't need to worry about where it comes from (fresh or from cache) - it's up to the caller to make sure they pass an `ISchemaLocater` that doesn't load schemas unnecessarily. And as far as I can tell that was **almost** happening except for passing the third argument to `getSchema`. However, due to what I believe is an API flaw, it was unnecessary.

In my opinion this confusion is caused by `SchemaContext` saying that it implements `ISchemaLocater` when it fact it doesn't. I think with changes in this PR it's even more apparent - `ISchemaLocater.getSchema` takes a required `SchemaContext` as the third argument and uses it to find referenced schemas. `SchemaContext.getSchema` doesn't use the third argument and instead looks for referenced schemas in its own cache. 

Consider this piece of code:
```ts
const context = new SchemaContext();
// add some schemas to `context`
const locater: ISchemaLocater = new SchemaContext();
const schemas = await locater.getSchema(key, matchType, context);
// referenced schemas in `context` won't be found
```
`getSchema` docs say that the third argument is going to be used for loading schema references, but it won't in this case.

To summarize, I think:
1. The change you made to our `getKoq` and `getFormattingProps` is okay.
2. `SchemaContext` should stop saying it implements `ISchemaLocater`.

Reply:
I see what you mean, but from API standpoint before your changes, I don't see why these methods couldn't have taken `ISchemaLocater`. All these functions care is to get a schema - they don't need to worry about where it comes from (fresh or from cache) - it's up to the caller to make sure they pass an `ISchemaLocater` that doesn't load schemas unnecessarily. And as far as I can tell that was **almost** happening except for passing the third argument to `getSchema`. However, due to what I believe is an API flaw, it was unnecessary.

In my opinion this confusion is caused by `SchemaContext` saying that it implements `ISchemaLocater` when in fact it doesn't. I think with changes in this PR it's even more apparent - `ISchemaLocater.getSchema` takes a required `SchemaContext` as the third argument and says it's going to use it to find referenced schemas. `SchemaContext.getSchema` doesn't use the third argument and instead looks for referenced schemas in its own cache. 

Consider this piece of code:
```ts
const context = new SchemaContext();
// add some schemas to `context`
const locater: ISchemaLocater = new SchemaContext();
const schemas = await locater.getSchema(key, matchType, context);
// referenced schemas in `context` won't be found
```
`getSchema` docs say that the third argument is going to be used for loading schema references, but it won't in this case.

To summarize, I think:
1. The change you made to our `getKoq` and `getFormattingProps` is okay.
2. `SchemaContext` should stop saying it implements `ISchemaLocater`.

Reply:
That is a good point, let me review why SchemaContext implements ISchemaLocater and see if I can remove it.

Reply:
I removed ISchemaLocater from SchemaContext and updated docs of ISchemaLocater to make its purpose more clear